switchboard/helpers.py

```diff
@@ -62,15 +62,19 @@ def _update_partial(self, old, new):
         for k, v in new.iteritems():
             old[k] = v
 
-    def update(self, spec, update):
+    def update(self, spec, update, upsert=False):
         current = self.find_one(spec)
         if not current:
-            return {
-                'err': None,
-                'n': 1,
-                'ok': 1.0,
-                'updatedExisting': True
-            }
+            if upsert:
+                spec.update(update)
+                return self.save(spec)
+            else:
+                return {
+                    'err': None,
+                    'n': 1,
+                    'ok': 1.0,
+                    'updatedExisting': True
```

Comment:
This class is a test helper for when a MongoDB isn't available; it's essentially a fake backend. It's been awhile since I wrote this code, but I suspect the odd data returned mimics what pymongo returns for a similar operation.